lib/proof/sn-comp-sector-seal/main.go

```diff
@@ -0,0 +1,211 @@
+package main
+
+import (
+	"encoding/binary"
+	"fmt"
+	"math/bits"
+	"os"
+	"path/filepath"
+
+	"github.com/minio/sha256-simd"
+	"golang.org/x/xerrors"
+
+	ffi "github.com/filecoin-project/filecoin-ffi"
+	"github.com/filecoin-project/filecoin-ffi/cgo"
+	"github.com/filecoin-project/go-commp-utils/zerocomm"
+	commcid "github.com/filecoin-project/go-fil-commcid"
+	"github.com/filecoin-project/go-state-types/abi"
+)
+
+func ReplicaId(sector abi.SectorNumber, ticket []byte, commd []byte) ([32]byte, error) {
+	// https://github.com/filecoin-project/rust-fil-proofs/blob/5b46d4ac88e19003416bb110e2b2871523cc2892/storage-proofs-porep/src/stacked/vanilla/params.rs#L758-L775
+
+	pi := [32]byte{9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9, 9}
+	porepID, err := abi.RegisteredSealProof_StackedDrg32GiBV1_1.PoRepID()
+	if err != nil {
+		return [32]byte{}, err
+	}
+
+	if len(ticket) != 32 {
+		return [32]byte{}, xerrors.Errorf("invalid ticket length %d", len(ticket))
+	}
+	if len(commd) != 32 {
+		return [32]byte{}, xerrors.Errorf("invalid commd length %d", len(commd))
+	}
+
+	var sectorID [8]byte
+	binary.BigEndian.PutUint64(sectorID[:], uint64(sector))
+
+	s := sha256.New()
+
+	// sha256 writes never error
+	_, _ = s.Write(pi[:])
+	_, _ = s.Write(sectorID[:])
+	_, _ = s.Write(ticket)
+	_, _ = s.Write(commd)
+	_, _ = s.Write(porepID[:])
+
+	return bytesIntoFr32Safe(s.Sum(nil)), nil
+}
+
+func bytesIntoFr32Safe(in []byte) [32]byte {
+	var out [32]byte
+	copy(out[:], in)
+
+	out[31] &= 0b0011_1111
+
+	return out
+}
+
+func main() {
+	outPath := os.Args[1]
```

Comment:
A  --help would give me an idea what this does.

Reply:
Added more explicit -h